pkg/utils/utils.go

```diff
@@ -458,3 +458,45 @@ func DumpServingCertSecret() error {
 
 	return nil
 }
+
+/*
+ComponentToCRDDirectory returns a map of component names to their corresponding CRD directory names.
+This mapping is used to skip CRD rendering for externally managed components.
+When NonOCP() returns true, the ClusterAPI component uses 'k8s' as its CRD directory.
+*/
+func ComponentToCRDDirectory() map[string]string {
+	var clusterAPICRDDir string
+	var clusterAPIProviderOACRDDir string
+	var clusterAPIProviderMetalCRDDir string
+
+	if DeployOnOCP() {
```

Comment:
IIUC soon we'll need to add other CAPI providers to k8s enablement.
As it stands, to test its enablement, we would need changes in the backplane operator.
Can we make this list dynamic (i.e. env var)?